### ISSUE TITLE
Expansion of ellipsis no longer faithful to what the user wrote

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#4778]
See https://github.com/agda/agda/commit/e531104ecdfd9efc1e76bf0e9b5f9690f31bcaab#r40133390
The expansion of an ellipsis
https://github.com/agda/agda/blob/528a9fd3788245d6d2f4d33d394e3305ae3f5bbc/test/interaction/ExpandEllipsis.agda#L27-L32
no longer produces the parent clause, but a dot-shuffled version:
https://github.com/agda/agda/blob/528a9fd3788245d6d2f4d33d394e3305ae3f5bbc/test/interaction/ExpandEllipsis.out#L11

**Comment by jespercockx** [comment]
@UlfNorell do you remember the reason for your changes in e531104ecdfd9efc1e76bf0e9b5f9690f31bcaab? It doesn't reference any issue as far as I can see.

**Comment by UlfNorell** [comment]
It was in the context of the new forcing translation. I don't remember exactly, but I believe the test suite failed checking with-functions due to incorrect hiding information.

**Comment by jespercockx** [comment]
Here are a couple of other related test cases:
- With further refinement:
```agda
postulate trustMe : {A : Set} (x y : A) → x ≡ y

test2 : ∀{n} → Fin n → Nat
test2 (fzero _) = 0
test2 {.(suc n)} (fsuc {n} i) with trustMe n zero
... | refl = {!.!}

-- Result after case splitting:
test2 {suc .0} (fsuc {.0} i) | refl = {!!}
```
- With variable instead of dot pattern:
```agda
test3 : ∀{n} → Fin n → Nat
test3 (fzero _) = 0
test3 {m} (fsuc {n} i) with Fin zero
... | q = {!.!}

-- Result after case splitting:
test3 {suc _} (fsuc {_} i) | q = {!!}
```
- With wildcard pattern:
```agda

test4 : ∀{n} → Fin n → Nat
test4 (fzero _) = 0
test4 {_} (fsuc {n} i) with Fin zero
... | q = {!.!}

-- Result after case splitting:
test4 {suc _} (fsuc {_} i) | q = {!!}
```
We should preserve not just dots, but also variables and wildcards. Basically, the only thing that should happen is replacing variables by dot patterns, but not by other things.

**Comment by jespercockx** [comment]
This is very frustrating. I was working on a general fix to the issues I posted above but it seems no matter what I change to the code for stripping the with patterns in `With.hs`, I always introduce some other problem. It's like playing whack-a-mode. I now believe the whole `stripWithClausePatterns` function should be burned to the ground and replaced with a much simpler and general approach that re-uses functions such as `updateProblemEqs` in `LHS.hs` (which essentially implements the same logic). However, since this will take some time and is likely to lead to at least some breaking changes, I won't do that right now. I will instead push a smaller fix for the problem in the OP and leave the other issues with expanding ellipsis that I posted above to future work.
